apps/users-service/src/framework/lambda.ts

```diff
@@ -90,7 +90,9 @@ export const http = <T>(
     query: event.queryStringParameters,
   } as Request;
 
-  const [err, res] = await Intercept(fn(request));
+  const [err, res] = await Intercept(
+    (async (): Promise<Response> => fn(request))(),
```

Comment:
Any particular reason for wrapping promises in a new `async` function?

Reply:
The definition of `fn` above returns `Response | Promise<Response>`. This is to prevent the `Response` case from breaking (`Intercept` can't deal with something that doesn't have a `.then`). If we don't actually need that we can change `fn` to return only `Promise<Response>` (probably better in fact)

Reply:
pushed